vendor_data.py

```diff
@@ -0,0 +1,156 @@
+    2. The subclass ´PrefVendor´, for the preferred vendors. They are assigned the status "Preferred".
+    3. The subclass ´BackupVendor´ for the back-up vendors. They are assigned the status "Back-up"
+"""
+
+import argparse
+import re
+import pandas as pd
+import os
+import pyinputplus as pyip
+
+class VendorData:
+    ProjectName = "Toyota MM24" #Class attribute no.1, the project the vendors will be working on
+    SourceLang = "English" #Class attribute no.2, the source language
+    CatTools = ["XTM", "Trados Studio", "MemoQ", "Memsource"] #Class attribute no.3, the list of possible Cat Tools
+    Status = "Potential"
+    
+    def __init__(self, VendorName, TargLang, WordRate, VendorMail = "", CatTool = "XTM"):
+        """ Instantiate
+        Args:
+            "VendorName" (str): First and last name of the vendor
+            "VendorMail" (str): the vendor's email address
+            "TargLang" (str): Language the vendor will be translating to
+            "WordRate" (float): Vendor's word rate for new words in Euro
+            "CatTool" (str): Vendor's preferred CAT-tool
+            """
+        self.VendorName = VendorName
+        self.TargLang = TargLang
+        self.WordRate = WordRate
+        self.VendorMail = VendorMail
+        self.CatTool = CatTool
+        
+        regex_mail = "^[a-z0-9]+[\._]?[a-z0-9]+[@]\w+[.]\w{2,3}$" #regex used to validate email
+            
+
+        """
+        Exceptions:
+            VendorName should be a string, if not a TypeError is raised.
+            TargLang should be a string, if not a TypeError is raised.
+            WordRate should be a float, if not a TypeError is raised.
+            WordRate should NOT be > 0.15, if not a ValueError is raised.
+            VendorMail should be a string, if not a TypeError is raised.
+            VendorMail should be an email, if not a ValueError is raised.
+            CatTool should be a string, if not a TypeError is raised.
+            CatTool should be one of the options in the list "CatTools", if not a ValueError is raised.
+        """
+        
+        if type(VendorName) !=str:
+            raise TypeError("VendorName should be a string!")
+        if type(TargLang) != str:
+            raise TypeError("TargLang should be a string!")
+        if type(WordRate) != float:
+            raise TypeError("WordRate should be a float!")
+        if WordRate:
+            if WordRate > 0.15:
+                raise ValueError("This vendor is too expensive, consider another one.")
+        elif WordRate == 0.00:
+            raise ValueError("Word rate cannot be 0.00.")
+        else:
+            self.WordRate = WordRate
+        if type(CatTool) != str:
+            raise TypeError("CatTool should be a string!")
+        if type(VendorMail) != str:
+            raise TypeError ("VendorMail should be a string!")
+        if VendorMail:   
+            if(re.search(regex_mail,VendorMail)):

```

Comment:
Good!

Reply:
(Just: `if type(VendorMail)` should be inside the `if VendorMail` part, because it fails if there is no `VendorMail`)

Reply:
Ok, now that I see a bit below, lines 78-81 can be replaced with `self.set_mail(VendorMail)`, and line 45 can be deleted.